pkg/codestruct/codestruct.go

```diff
@@ -0,0 +1,39 @@
+package codestruct
+
+import (
+	"go/ast"
+)
+
+type Weights struct {
+	Func      uint `xml:"func"`
```

Comment:
Btw why do we use `uint` as weights everywhere? Won't using `float64` make tweaking them easier (by allowing changing them more preciesly)?